### PR TITLE
Correct toml/FFJORD

### LINKED ISSUE #47: FFJORD
```julia
using OrdinaryDiffEq
using Distributions
using Flux, DiffEqFlux, Tracker

# Neural Network
nn = Chain(Dense(1,1,tanh))
p = Tracker.data(DiffEqFlux.destructure(nn))
DiffEqFlux.restructure(nn,p)([1.0])
tspan = Float32.((0.0, 10.0))
function cnf(du,u,p,t)
  z = @view u[1:end-1]
  m = DiffEqFlux.restructure(nn,p)
  du[1:end-1] = m(z)
  fz, back = Tracker.forward(z->m(z),z')
  e = randn(size(z)[1])
  eJ = back(e)[1]
  eJe = Tracker.data((eJ .* e)[1])
  du[end] = -eJe
end

Tracker.collectmemaybe(z)

prob = ODEProblem(cnf,nothing,tspan,nothing)

p = param(Float32[0.0, 0.0]) # Initial Parameter Vector
params = Params([p])

function predict_adjoint(x)
    diffeq_adjoint(p,prob,Tsit5(),u0=[x;false],
                   saveat=0.0:0.1:10.0,
                   sensealg=DiffEqFlux.SensitivityAlg(quad=false,
                                backsolve=true,autojacvec=true))
end

function loss_adjoint(xs)
    pz = Normal(0.0, 1.0)
    preds = [predict_adjoint(x)[:,end] for x in xs]
    z = [pred[1] for pred in preds] # TODO better slicing
    delta_logp = [pred[2] for pred in preds]

    logpz = logpdf.(pz, z)
    logpx = logpz - delta_logp
    loss = -mean(logpx)
end

opt = ADAM(0.1)

raw_data = [Float32[rand(Normal(2.0, 0.1)) for i in 1:100]]
data = Iterators.repeated(raw_data, 100);

Flux.train!(loss_adjoint, params, data, opt)

# check whether it looks standard normal
using Plots

preds = [predict_adjoint(r)[:,end] for r in raw_data[1]];
```

### LINKED ISSUE #282: Library-tized FFJORD
@ChrisRackauckas: base distributions and the method of calculation of the trace of the Jacobian can be changed by the user.
```Julia

using Distributions
using DifferentialEquations
using Flux, DiffEqFlux
using Zygote, Tracker
using Plots

##
##Library-tized version
##

abstract type CNFLayer <: Function end
Flux.trainable(m::CNFLayer) = (m.p,)

struct FFJORD{M,P,RE,ContinuousUnivariateDistribution,Bool,T,A,K} <: CNFLayer
    model::M
    p::P
    re::RE
    baseDist::ContinuousUnivariateDistribution
    MonteCarlo::Bool
    tspan::T
    args::A
    kwargs::K

    function FFJORD(model,tspan,args...;p = nothing,baseDist=nothing,MonteCarlo=nothing,kwargs...)
        _p,re = Flux.destructure(model)
        if p === nothing
            p = _p
        end
        if baseDist === nothing
            baseDist = Normal(0,1)
        end
        if MonteCarlo === nothing
            MonteCarlo = false
        end
        new{typeof(model),typeof(p),typeof(re),typeof(baseDist),
            typeof(MonteCarlo),typeof(tspan),typeof(args),typeof(kwargs)}(
            model,p,re,baseDist,MonteCarlo,tspan,args,kwargs)
    end
end

function ffjord(du,u,p,t,re,MonteCarlo,e)
    z = @view u[1:end-1]
    m = re(p)
    fz, back = Zygote.pullback(m,z')
    if MonteCarlo
        eJ = back(e)[1]
        jac = -(eJ.*e)[1]
    else
        jac = -sum(back(1)[1])
    end
    du[1:end-1] .= m(z)
    du[end] = jac
end

function (n::FFJORD)(x,p=n.p,MonteCarlo=n.MonteCarlo)
    e = MonteCarlo ? randn(Float32,size(p)[1]) : nothing
    ffjord_ = (du,u,p,t)->ffjord(du,u,p,t,n.re,MonteCarlo,e)
    prob = ODEProblem{true}(ffjord_,nothing,n.tspan,nothing)
    pred = concrete_solve(prob,Tsit5(),[x,0f0],p,n.args...;n.kwargs...)[:,end]
    pz = n.baseDist
    z = pred[1]
    delta_logp = pred[2]
    logpz = logpdf(pz, z)
    logpx = logpz - delta_logp
    return logpx
end

##
##Example of use with standard base distribution and MonteCarlo = true
##

nn = Chain(Dense(1, 1, tanh))
data_train = [Float32(rand(Beta(7,7))) for i in 1:100]
tspan = (0.0,10.0)
ffjord_test_mc = FFJORD(nn,tspan,MonteCarlo=true)

function loss_adjoint(θ)
    logpx = [ffjord_test_mc(x,θ) for x in data_train]
    loss = -mean(logpx)
end

function cb(p, l)
    @show p, l
    false
end

res = DiffEqFlux.sciml_train(loss_adjoint, ffjord_test_mc.p,
                                        ADAM(0.1), cb = cb,
                                        maxiters = 100)

θopt = res.minimizer
@show θopt
data_validate = [Float32(rand(Beta(7,7))) for i in 1:100]

#Use deterministic for plotting
preds_mc = [(r, exp(ffjord_test_mc(r, θopt,false))) for r in data_validate]
pl = scatter([(r, pdf(Beta(7,7),r)) for r in data_validate],label="Beta(7,7)")
scatter!(preds_mc,label="Neural Network")
display(pl)

##
##Example of use with alternate baseDist and MonteCarlo=false
##

nn = Chain(Dense(1, 1, tanh))
data_train = [Float32(rand(Normal(6.0,0.7))) for i in 1:10]
tspan = (0.0,10.0)
ffjord_test = FFJORD(nn,tspan,baseDist=Normal(0,2),MonteCarlo=false)

function loss_adjoint(θ)
    logpx = [ffjord_test(x,θ) for x in data_train]
    loss = -mean(logpx)
end

function cb(p, l)
    @show p, l
    false
end

res = DiffEqFlux.sciml_train(loss_adjoint, ffjord_test.p,
                                          ADAM(0.1), cb = cb,
                                          maxiters = 100)

data_validate = [Float32(rand(Normal(6.0,0.7))) for i in 1:100]
θopt = res.minimizer
preds = [(r, exp(ffjord_test(r, θopt))) for r in data_validate]
pl = scatter([(r, pdf(Normal(6.0,0.7),r)) for r in data_validate],label="Normal(6.0,0.7)")
scatter!(preds,label="Neural Network")
display(pl)


```

### PR DESCRIPTION
**Pull request by d-netto**
branch of #287

Closes #47 and closes #282

### CODE REVIEW

File: docs/Project.toml [hunk 1]
```diff
@@ -6,6 +6,7 @@ Documenter = "0.24.2"
 
 [extras]
 Distributions = "31c24e10-a181-5473-b8eb-7969acd0382f"
+Distances = "b4f34e82-e78d-54a5-968a-f98e89d6e8f7"
```
Comment (ChrisRackauckas) [inline docs/Project.toml:9]: the documentation Project.toml doesn't need to be edited.

Comment (d-netto) [reply docs/Project.toml:9]: Ok, I'll change it.

**Comment by d-netto** [comment]
Throws "UndefVarError: FFJORD not defined".

 Stacktrace:
736   [1] top-level scope at /home/travis/build/SciML/DiffEqFlux.jl/test/cnf_test.jl:12
737   [2] include(::Module, ::String) at ./Base.jl:377
738   [3] include(::String) at /home/travis/.julia/packages/SafeTestsets/A83XK/src/SafeTestsets.jl:23
739   [4] top-level scope at /home/travis/build/SciML/DiffEqFlux.jl/test/runtests.jl:17
740   [5] top-level scope at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Test/src/Test.jl:1113
741   [6] top-level scope at /home/travis/build/SciML/DiffEqFlux.jl/test/runtests.jl:17
742   [7] eval(::Module, ::Any) at ./boot.jl:331
743   [8] top-level scope at /home/travis/.julia/packages/SafeTestsets/A83XK/src/SafeTestsets.jl:23
744   [9] top-level scope at /home/travis/build/SciML/DiffEqFlux.jl/test/runtests.jl:17
745   [10] top-level scope at util.jl:175
746   [11] include(::String) at ./client.jl:439
747   [12] top-level scope at none:6
748   [13] eval(::Module, ::Any) at ./boot.jl:331
749   [14] exec_options(::Base.JLOptions) at ./client.jl:264
750   [15] _start() at ./client.jl:484
751  in expression starting at /home/travis/build/SciML/DiffEqFlux.jl/test/cnf_test.jl:12
752

**Comment by ChrisRackauckas** [comment]
add `export FFJORD` to the top level of the package (see the other exports)

**Comment by d-netto** [comment]
Throws

ERROR: LoadError: LoadError: UndefVarError: ContinuousMultivariateDistribution not defined
8
Stacktrace:
9
 [1] top-level scope at /home/runner/work/DiffEqFlux.jl/DiffEqFlux.jl/src/ffjord.jl:4
10
 [2] include(::Module, ::String) at ./Base.jl:377
11
 [3] include(::String) at /home/runner/work/DiffEqFlux.jl/DiffEqFlux.jl/src/DiffEqFlux.jl:1
12
 [4] top-level scope at /home/runner/work/DiffEqFlux.jl/DiffEqFlux.jl/src/DiffEqFlux.jl:68
13
 [5] include(::Module, ::String) at ./Base.jl:377
14
 [6] top-level scope at none:2
15
 [7] eval at ./boot.jl:331 [inlined]
16
 [8] eval(::Expr) at ./client.jl:449
17
 [9] top-level scope at ./none:3
18
in expression starting at /home/runner/work/DiffEqFlux.jl/DiffEqFlux.jl/src/ffjord.jl:4
19
in expression starting at /home/runner/work/DiffEqFlux.jl/DiffEqFlux.jl/src/DiffEqFlux.jl:68
20
ERROR: LoadError: Failed to precompile DiffEqFlux [aae7a2af-3d4f-5e19-a356-7da93b79d9d0] to /home/runner/.julia/compiled/v1.4/DiffEqFlux/BdO4p_aDECi.ji.
21
Stacktrace:
22
 [1] error(::String) at ./error.jl:33
23
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1272
24
 [3] _require(::Base.PkgId) at ./loading.jl:1029
25
 [4] require(::Base.PkgId) at ./loading.jl:927
26
 [5] require(::Module, ::Symbol) at ./loading.jl:922
27
 [6] include(::Module, ::String) at ./Base.jl:377
28
 [7] exec_options(::Base.JLOptions) at ./client.jl:288
29
 [8] _start() at ./client.jl:484
30
in expression starting at /home/runner/work/DiffEqFlux.jl/DiffEqFlux.jl/docs/make.jl:1

**Comment by d-netto** [comment]
I though multivariate was defined in distributions.

**Comment by d-netto** [comment]
> @ChrisRackauckas I though ContinuousMultivariateDistribution was defined in Distributions.jl as well.

Resolved.

**Comment by d-netto** [comment]
@ChrisRackauckas: it's now throwing an error in cnf_test.jl despite "using Distributions".

LoadError: UndefVarError: Normal not defined
735  Stacktrace:
736   [1] top-level scope at /home/travis/build/SciML/DiffEqFlux.jl/test/cnf_test.jl:13
737   [2] include(::Module, ::String) at ./Base.jl:377
738   [3] include(::String) at /home/travis/.julia/packages/SafeTestsets/A83XK/src/SafeTestsets.jl:23
739   [4] top-level scope at /home/travis/build/SciML/DiffEqFlux.jl/test/runtests.jl:17
740   [5] top-level scope at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Test/src/Test.jl:1113
741   [6] top-level scope at /home/travis/build/SciML/DiffEqFlux.jl/test/runtests.jl:17
742   [7] eval(::Module, ::Any) at ./boot.jl:331
743   [8] top-level scope at /home/travis/.julia/packages/SafeTestsets/A83XK/src/SafeTestsets.jl:23
744   [9] top-level scope at /home/travis/build/SciML/DiffEqFlux.jl/test/runtests.jl:17
745   [10] top-level scope at util.jl:175
746   [11] include(::String) at ./client.jl:439
747   [12] top-level scope at none:6
748   [13] eval(::Module, ::Any) at ./boot.jl:331
749   [14] exec_options(::Base.JLOptions) at ./client.jl:264
750   [15] _start() at ./client.jl:484
751  in expression starting at /home/travis/build/SciML/DiffEqFlux.jl/test/cnf_test.jl:13

**Comment by d-netto** [comment]
> @ChrisRackauckas: it's now throwing an error in cnf_test.jl despite "using Distributions".
> 
> LoadError: UndefVarError: Normal not defined
> 735 Stacktrace:
> 736 [1] top-level scope at /home/travis/build/SciML/DiffEqFlux.jl/test/cnf_test.jl:13
> 737 [2] include(::Module, ::String) at ./Base.jl:377
> 738 [3] include(::String) at /home/travis/.julia/packages/SafeTestsets/A83XK/src/SafeTestsets.jl:23
> 739 [4] top-level scope at /home/travis/build/SciML/DiffEqFlux.jl/test/runtests.jl:17
> 740 [5] top-level scope at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Test/src/Test.jl:1113
> 741 [6] top-level scope at /home/travis/build/SciML/DiffEqFlux.jl/test/runtests.jl:17
> 742 [7] eval(::Module, ::Any) at ./boot.jl:331
> 743 [8] top-level scope at /home/travis/.julia/packages/SafeTestsets/A83XK/src/SafeTestsets.jl:23
> 744 [9] top-level scope at /home/travis/build/SciML/DiffEqFlux.jl/test/runtests.jl:17
> 745 [10] top-level scope at util.jl:175
> 746 [11] include(::String) at ./client.jl:439
> 747 [12] top-level scope at none:6
> 748 [13] eval(::Module, ::Any) at ./boot.jl:331
> 749 [14] exec_options(::Base.JLOptions) at ./client.jl:264
> 750 [15] _start() at ./client.jl:484
> 751 in expression starting at /home/travis/build/SciML/DiffEqFlux.jl/test/cnf_test.jl:13

I'll try to to Distributions to DiffEqFlux.jl

File: src/DiffEqFlux.jl [hunk 2]
```diff
@@ -65,15 +65,16 @@ ZygoteRules.@adjoint ZygoteRules.literal_getproperty(d::ForwardDiff.Dual{T}, ::V
...
 include("fast_layers.jl")
 include("neural_de.jl")
 include("require.jl")
 
 
 export diffeq_fd, diffeq_rd, diffeq_adjoint
-export NeuralODE, NeuralDSDE, NeuralSDE, NeuralCDDE, NeuralDAE, NeuralODEMM
-export neural_ode, neural_ode_rd
+export FFJORD, NeuralODE, NeuralDSDE, NeuralSDE, NeuralCDDE, NeuralDAE, NeuralODEMM
+export ffjord, neural_ode, neural_ode_rd
```
Comment (ChrisRackauckas) [inline src/DiffEqFlux.jl:77]: this is internal so it doesn't need to be exported.

Comment (ChrisRackauckas) [inline src/DiffEqFlux.jl:77]:
```suggestion
export neural_ode, neural_ode_rd
```

**Comment by ChrisRackauckas** [comment]
Looks great. I'll merge when tests pass.

@jessebett you might want to take a look at this.

**Pull request merged**